### PR TITLE
Android: perform DNS over TCP when UDP is not available

### PR DESCRIPTION
**Pull request by alalamav**
* Sends a truncated response to DNS queries over UDP so that they are retried over TCP when the network or remote server do not support UDP forwarding.
  * Accomplished by enabling DNS resolution over TCP in tun2socks when transparent DNS is set, and no UDP relay or DNS resolver are specified. Note that this change is fully compatible with Windows.
* Performs the UDP connectivity check on network changes.

### CODE REVIEW

File: third_party/badvpn/tun2socks/tun2socks.c [hunk 1]
```diff
@@ -284,6 +284,11 @@ static void udp_send_packet_to_device (void *unused, BAddr local_addr, BAddr rem
 //==== OUTLINE ====
 #ifndef BADVPN_USE_WINAPI
 
+// Returns whether DNS resolution over UDP is supported.
```
Comment (trevj) [inline third_party/badvpn/tun2socks/tun2socks.c]: Cool! Let's update `third_party/badvpn/METADATA` to help track our own changes.

**Comment by trevj** [comment]
@alalamav Perhaps update the PR description to reflect the changes to tun2socks?

File: third_party/badvpn/tun2socks/tun2socks.c (hunk 1 above)
Comment (alalamav) [reply third_party/badvpn/tun2socks/tun2socks.c]: Done in e23932a.

File: cordova-plugin-outline/android/java/org/outline/vpn/VpnTunnelService.java [hunk 2]
```diff
@@ -142,58 +144,65 @@ public void onDestroy() {
...
     if (connectionId == null || config == null) {
       throw new IllegalArgumentException("Must provide a connection ID and configuration.");
-    } else if (connectionId.equals(activeConnectionId)) {
-      LOG.info(
-          String.format(Locale.ROOT, "Already running tunnel for connection ID %s", connectionId));
-      broadcastVpnStart(OutlinePlugin.ErrorCode.NO_ERROR); // Start is idempotent.
-      return;
-    } else if (activeConnectionId != null) {
+    }
+    boolean isRestart = activeConnectionId != null;
```
[the remark this answers is not in the document]
Comment (alalamav) [reply cordova-plugin-outline/android/java/org/outline/vpn/VpnTunnelService.java]: Done in e23932a.

**Comment by alalamav** [comment]
Thanks for the review @trevj. I updated the PR description and tun2socks' METADATA (also added past changes). PTAL.

**Pull request merged**